### ISSUE TITLE
rvierdiiev - BufferBinaryOptions.createFromRouter is not checking for pause

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2022-11-buffer-judging#5]
rvierdiiev

medium

# BufferBinaryOptions.createFromRouter is not checking for pause

## Summary
Because `BufferBinaryOptions.createFromRouter` is not checking for pause it's still possible to create new option after BufferBinaryOptions contract was paused.
## Vulnerability Detail
Function BufferBinaryOptions.toggleCreation is created for [pausing/unpausing](https://github.com/sherlock-audit/2022-11-buffer/blob/main/contracts/contracts/core/BufferBinaryOptions.sol#L91-L97) contract.
When contract is paused that means that no new options should be created.
In BufferBinaryOptions.runInitialChecks there is check if contract is paused. If it is, then you can't initiateTrade.
https://github.com/sherlock-audit/2022-11-buffer/blob/main/contracts/contracts/core/BufferBinaryOptions.sol#L275-L285
```solidity
    function runInitialChecks(
        uint256 slippage,
        uint256 period,
        uint256 totalFee
    ) external view override {
        require(!isPaused, "O33");
        require(slippage <= 5e2, "O34"); // 5% is the max slippage a user can use
        require(period >= config.minPeriod(), "O21");
        require(period <= config.maxPeriod(), "O25");
        require(totalFee >= config.minFee(), "O35");
    }
```

BufferBinaryOptions.runInitialChecks [is called](https://github.com/sherlock-audit/2022-11-buffer/blob/main/contracts/contracts/core/BufferRouter.sol#L82) by BufferRouter.initiateTrade. That means that if BufferBinaryOptions is paused then user can't initiate new trade.

But still there is a chance to create option when BufferBinaryOptions already paused.
Consider example.
1.User call BufferRouter.initiateTrade and queues his trade.
2.BufferBinaryOptions is paused.
3.BufferRouter.resolveQueuedTrades is called by keeper.
4.Because no more checks for pausing, option is created.
## Impact
There is possibility to create option even if BufferBinaryOptions is paused
## Code Snippet
Provided above
## Tool used

Manual Review

## Recommendation
Add pause check also to `BufferBinaryOptions.createFromRouter` function or to `BufferBinaryOptions.isStrikeValid` function.

**Comment by rvierdiyev** [comment]
Escalate for 1 USDC.
Even that this will be only possible during 1 minute enough for some users to lose(freeze) their funds in case if BufferBinaryOptions will not be unpaused(or during time when it will be paused).
When the contract is paused it should not allow creating options.
I think that severity should be medium as it leads to users funds lock.
The fact that protocol team fixed that issue also says that the problem is real.

**Comment by 0x00052** [comment]
Fixed in [PR#4](https://github.com/bufferfinance/Buffer-Protocol-v2/pull/4)

Changes look good. BufferBinaryOptions#checkParams now checks if in contract is paused, which prevents options from being resolved (opened) by BufferRouter when option contract is paused.
